Alignment/CommonAlignmentMonitor/plugins/AlignmentStats.cc

```diff
@@ -88,7 +88,7 @@ void AlignmentStats::beginJob() {  // const edm::EventSetup &iSetup
 void AlignmentStats::analyze(const edm::Event &iEvent, const edm::EventSetup &iSetup) {
   //load list of detunits needed then in endJob
   if (!trackerGeometry_) {
-    trackerGeometry_ = std::make_unique<TrackerGeometry>(iSetup.getData(esTokenTkGeo_));
+    trackerGeometry_ = &iSetup.getData(esTokenTkGeo_);
```

Comment:
I see `trackerGeometry_` is used in `endJob()`, at which stage the framework has the freedom to have deleted the EventSetup data product whose address is taken here.

Reply:
which is I think the reason it was done this way. What's the suggestion from the framework point of view?

Reply:
Ideal solution would be to make the copy constructor of `TrackerGeometry` to behave properly (the copy assignment could probably still be left deleted, if desired).

Reply:
No longer relevant as it has been descoped to another PR.